crm_location/models/crm_lead.py

```diff
@@ -45,11 +26,8 @@ def on_change_city(self):
     )
 
     @api.multi
-    def on_change_partner_id(self, partner_id):
-        res = super(CrmLead, self).on_change_partner_id(partner_id)
-        if 'value' not in res:
-            res['value'] = {}
+    def _onchange_partner_id_values(self, partner_id):
```

Comment:
This old onchange doesn't work for sure. Please use new API onchange (you can create a new method for not having to overwrite super):
```
@api.onchange('partner_id')
def onchange_partner_id_crm_location(self):
    if self.partner_id:
        self.location_id = self.partner_id.zip_id.id
```